recompose-idea/src/main/kotlin/recompose/plugin/copypaste/RecomposeCopyPasteProcessor.kt

```diff
@@ -58,14 +59,28 @@ class RecomposeCopyPasteProcessor : CopyPastePostProcessor<TextBlockTransferable
     override fun extractTransferableData(
         content: Transferable
     ): List<TextBlockTransferableData> {
-        if (content.isDataFlavorSupported(CopiedXMLCode.DATA_FLAVOR)) {
-            // There's some matching data in this paste, let's return it to process it.
-            return listOf(
-                content.getTransferData(CopiedXMLCode.DATA_FLAVOR) as TextBlockTransferableData
-            )
-        }
 
-        return emptyList()
+        return if (content.isDataFlavorSupported(DataFlavor.stringFlavor)) {
+            val text = content.getTransferData(DataFlavor.stringFlavor) as String
+            val xmlPreamble = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>"
+
+            when {
+                content.isDataFlavorSupported(CopiedXMLCode.DATA_FLAVOR) -> {
```

Comment:
Do we ever get here if this block is wrapped in `DataFlavor.stringFlavor`? Is that an extension of the string flavor?

Reply:
Yes, we can only get here when the content has one flavour with DataFlavor.stringFlavor, see row 63. 
I'm first checking if the pasted content is a string and not e.g. an image file. Then i'm checking if the content also has the dataflavor of CopiedXMLCode. If it's not, i'm checking if it looks like XML.

Reply:
Alright 👍